addons/hr_expense/models/hr_expense.py

```diff
@@ -152,9 +153,15 @@ def _compute_state(self):
     @api.depends('quantity', 'unit_amount', 'tax_ids', 'currency_id')
     def _compute_amount(self):
         for expense in self:
-            expense.untaxed_amount = expense.unit_amount * expense.quantity
-            taxes = expense.tax_ids.compute_all(expense.unit_amount, expense.currency_id, expense.quantity, expense.product_id, expense.employee_id.user_id.partner_id)
-            expense.total_amount = taxes.get('total_included')
+            if expense.unit_amount:
+                expense.untaxed_amount = expense.unit_amount * expense.quantity
+                taxes = expense.tax_ids.compute_all(expense.unit_amount, expense.currency_id, expense.quantity, expense.product_id, expense.employee_id.user_id.partner_id)
+                expense.total_amount = taxes.get('total_included')
+            else:
+                # the taxes should be "Included In Price", as the entered
+                # total_amount includes all the taxes already
+                taxes = expense.tax_ids.compute_all(expense.total_amount, expense.currency_id, expense.quantity, expense.product_id, expense.employee_id.user_id.partner_id)
+                expense.total_amount = taxes.get('total_included')
```

Comment:
Yes, you do have a point. But if we remove the else part, it might influence other variables. Because it won't compute other variables based on the taxes and total amount. If we have an account move later (for taxes maybe), it might lead to an incorrect result. I added the else part just to make sure all the values are correct so we won't have some bugs difficult to trace later :) Do you think it's a reasonable explanation? :thinking:

Reply:
But here as the taxes are price_included, the existing total_amount will already be equal to what will be computed. IIRC when posting the journal entries it will anyway compute everything again  :-)

Reply:
Ahhh, indeed, I tested it and it worked fine without the else part. I will change the code then. Thanks! :smile: